storage/indexer/src/db_indexer.rs

```diff
@@ -436,6 +479,66 @@ impl DBIndexer {
         Ok(version)
     }
 
+    fn should_translate(v2: &ContractEventV2) -> bool {
+        matches!(
+            v2.type_tag().to_canonical_string().as_str(),
+            "00000000000000000000000000000001::coin::CoinDeposit"
+                | "00000000000000000000000000000001::coin::CoinWithdraw"
+        )
+    }
+
+    fn get_resource(&self, address: &AccountAddress, struct_tag_str: &str) -> Result<StateValue> {
+        let state_view = self
+            .main_db_reader
+            .latest_state_checkpoint_view()
+            .expect("Failed to get state view");
+
+        let struct_tag = StructTag::from_str(struct_tag_str)?;
+        let state_key = StateKey::resource(address, &struct_tag)?;
+        let state_value = state_view
+            .get_state_value(&state_key)
+            .ok()
+            .unwrap()
+            .unwrap();
+        Ok(state_value)
+    }
+
+    fn translate_event_v2_to_v1(&self, v2: &ContractEventV2) -> ContractEventV1 {
+        match v2.type_tag().to_canonical_string().as_str() {
+            "00000000000000000000000000000001::coin::CoinDeposit" => {
+                let coin_deposit = CoinDeposit::try_from_bytes(v2.event_data()).unwrap();
+                let struct_tag_str = format!("0x1::coin::CoinStore<{}>", coin_deposit.coin_type());
+                // We can use `DummyCoinType` as it does not affect the correctness of deserialization.
+                let state_value = self
+                    .get_resource(&AccountAddress::ONE, &struct_tag_str)
+                    .unwrap();
+                let coin_store_resource: CoinStoreResource<DummyCoinType> =
+                    bcs::from_bytes(state_value.bytes()).unwrap();
+
+                let key = *coin_store_resource.deposit_events().key();
+                let sequence_number = self.indexer_db.get_next_sequence_number(&key).unwrap();
```

Comment:
two things:

1. you need to have some mechanism to ensure that once we see the v2 events, we cannot emit the corresponding v1 event anymore. 
2. you don't write to db after every txn, so basically for each batch you will get the same value from db, you need to maintain the value in memory until you commit it to db.

Reply:
1. Yes. As soon as we start emitting V2 events from the Aptos Framework, we will stop emitting V1 events. It's done in this PR: https://github.com/aptos-labs/aptos-core/pull/14432
2. `event_sequence_number_cache` has been added to `DBIndexer` for that reason. We now use it to address the issue.